core/src/main/scripts/importer/validateData.py

```diff
@@ -2573,6 +2579,112 @@ def parse_q_value(self, value, column_number):
+    
+    REQUIRED_HEADERS = ['geneset_id']
+    def __init__(self, *args, **kwargs):
+        super(GsvaWiseFileValidator, self).__init__(*args, **kwargs)
+        self.geneset_ids = []
+
+    def checkHeader(self, cols):
+        """Validate the header and read sample IDs from it.
+
+        Return the number of fatal errors.
+        """
+        num_errors = super(GsvaWiseFileValidator, self).checkHeader(cols)
+
+        global GSVA_SAMPLE_IDS
+
+        if GSVA_SAMPLE_IDS != None:
+            if self.cols != GSVA_SAMPLE_IDS:
+                self.logger.error('Headers from score and p-value files are different',
+                                  extra={'line_number': self.line_number})
+                num_errors += 1
+        else:
+            GSVA_SAMPLE_IDS = self.cols
+            
+        return num_errors
+
+    def parseFeatureColumns(self, nonsample_col_vals):
+
+        """Check the `geneset_id` column."""  
+        
+        global GSVA_GENESET_IDS
+        
+        geneset_id = nonsample_col_vals[0].strip()      
+        #Check if gene set is present
+        if geneset_id == '':
+            # Validator already gives warning for this in checkLine method   
+            pass
+        # Check if gene set contains whitespace    
+        elif ' ' in geneset_id:
+            self.logger.error("Whitespace found in `geneset_id`",
+                              extra={'line_number': self.line_number,
+                                     'cause': geneset_id})     
+        # Check if gene set is in database
+        elif self.portal.geneset_id_list is not None and geneset_id not in self.portal.geneset_id_list:
+            self.logger.warning("Gene set not found in database, please make sure "
+                                "to import gene sets prior to study loading",
+                              extra={'line_number': self.line_number, 'cause': geneset_id})      
+        else:
+            # Check if this is the second GSVA data file
+            if GSVA_GENESET_IDS != None:
+                # Check if gene set is in the first GSVA file
+                if not geneset_id in GSVA_GENESET_IDS:
+                    self.logger.error('Gene sets in GSVA score and p-value files are not equal',
+                                  extra={'line_number': self.line_number})
+            self.geneset_ids.append(geneset_id) 
+        return geneset_id
+    
+    def onComplete(self):
+        global GSVA_GENESET_IDS
+        
+        if GSVA_GENESET_IDS == None:
+            GSVA_GENESET_IDS = self.geneset_ids
+        else:
+            # Check if geneset ids are the same 
+            if not GSVA_GENESET_IDS == self.geneset_ids:
+                self.logger.error(
+                    'First columns of GSVA score and p-value files are not equal')
+        super(GsvaWiseFileValidator, self).onComplete()
+
+
+class GsvaScoreValidator(GsvaWiseFileValidator):
+    """Validator for files containing scores per gegene setrom GSVA algorithm.
```

Comment:
Indeed, thanks. I've updated some of the text because also GSVA-like scores such as ssGSEA are possible.